packages/api-rest/src/utils/createCancellableOperation.ts

```diff
@@ -58,9 +59,10 @@ export function createCancellableOperation(
 			return response;
 		} catch (error: any) {
 			const abortSignal = internalPostAbortSignal ?? publicApisAbortSignal;
+			const message = abortSignal.reason ?? abortReason;
```

Comment:
This should be:

```suggestion
			const message = abortReason ?? abortSignal.reason;
```

With the JSDOM update, it is ~more correct~ as it sets the default `signal.reason` to be a DOMException, hence enters the if statement below and triggers the error of setting readonly properties. 

However it does not fully support the spec because it ignores the `controller.abort(reason)` bellow. 

By switching the preference, we always set the `reason` supplied by the users if the runtime's `signal.reason` contradicts what users supply. 

Original code works in Browsers and RN because the `cotroller.abort(reason)` and readonly `signal.reason` are supported at the same time.

Reply:
Thank you for looking into this. I lacked the context to be confident about this change - updated as suggested